src/toil_scripts/lib/files.py

```diff
@@ -0,0 +1,81 @@
+from contextlib import closing
+import os
+import errno
+import tarfile
+import shutil
+
+
+def mkdir_p(path):
+    """
+    It is Easier to Ask for Forgiveness than Permission
+
+    :param str path: path to directory to create
+    """
+    try:
+        os.makedirs(path)
+    except OSError as exc:
+        if exc.errno == errno.EEXIST and os.path.isdir(path):
+            pass
+        else:
+            raise
+
+
+def tarball_files(tar_name, work_dir='.', fpaths=None, uuid=None):
+    """
+    Creates a tarball from a group of files
+
+    :param str tar_name: Name of tarball
+    :param str work_dir: Current working directory
+    :param list fpaths: File paths to include in the tarball
+    :param str uuid: UUID to stamp files with
+    """
+    if fpaths is None:
+        fpaths = []
+    with tarfile.open(os.path.join(work_dir, tar_name), 'w:gz') as f_out:
+        for fpath in fpaths:
+            if uuid:
+                f_out.add(fpath, arcname=uuid + '.' + os.path.basename(fpath))
```

Comment:
Remove '.' then people can chose whatever prefix they want.

There too much duplicated code in the if and else branches.